content/docs/03-ai-sdk-core/15-tools-and-tool-calling.mdx

```diff
@@ -693,15 +693,34 @@ const mcpClient = await createMCPClient({
 
 #### Closing the MCP Client
 
-After initialization, always close the MCP client when you're done to prevent resource leaks. Use try/finally or cleanup functions in your framework:
+After initialization, always close the MCP client after the response is finished to prevent resource leaks. 
+
+When streaming responses, close the client when the LLM response has finished. For example, when using `streamText`, you can use the `onFinish` callback:
+
+```typescript
+const mcpClient = await experimental_createMCPClient({
+  // ...
+});
+
+const result = await streamText({
+  model: openai('gpt-4o'),
+  tools: mcpClient.tools(),
+  prompt: 'What is the weather in Brooklyn, New York?',
+  onFinish: async () => {
+    await mcpClient.close();
+  },
+});
+```
+
+When generating responses without streaming, use try/finally or cleanup functions in your framework:
```

Comment:
Are there cases where we want to have a long running client that's kept open and only shut down when the app is terminated (e.g. for command line apps / agents)?